Cargo.toml

```diff
@@ -30,7 +30,7 @@ include = [
     "Cargo.toml",
     "LICENSE.TXT",
 ]
-edition = "2021"
+edition = "2024"
```

Comment:
my only concern is if this will force users of slparser to use newer versions of the rust compiler (as in once we release sqlparser with this change, will it require the use of rust 1.85?)

I haven't studied the implications yet